piksi_multi_cpp/src/receiver/receiver_position.cc

```diff
@@ -4,6 +4,26 @@ namespace piksi_multi_cpp {
 
 ReceiverPosition::ReceiverPosition(const ros::NodeHandle& nh,
                                    const std::shared_ptr<Device>& device)
-    : Receiver(nh, device) {}
+    : Receiver(nh, device),
+      // attach udp receiver with this calss as consumer.
+      udp_receiver_(ObservationsConsumer::Ptr(this)) {}
```

Comment:
One thing I don't understand is are we ever expecting to have more than one consumer per receiver?

Reply:
in this case actually not, so changed it!